book/src/prefab.md

```diff
@@ -0,0 +1,212 @@
+# Prefabs
+
+A `Prefab` in `Amethyst` is at the core a simple list of future entities, where each entry in 
+the list consists of two pieces of optional data:
+
+* a parent index that refers to a different entry in the list
+* a data collection implementing the trait `PrefabData`
+
+To instantiate a `Prefab`, we put a `Handle<Prefab<T>>` on an `Entity`. The `Entity` we put 
+the `Handle` on is referred to as the main `Entity`, and the first entry in the list inside a 
+`Prefab` refers to this `Entity`. All other entries in the list will spawn a new `Entity` on 
+instantiation. 
+
+NOTE: This means that we currently cannot target multiple existing entities from a single `Prefab`.
```

Comment:
What does it take to remove this restriction?  It's pretty easy to run into this in say an RTS game.

Reply:
I don't know at this point, I've not thought about it a lot. You'll need some kind of persistent addressing scheme for the entities, probably `Marker` from `saveload`. It is also fairly simple to cause UB with that, because you can easily create hierarchy loops.